python/elasticbeanstalk/elasticbeanstalk-bg-pipeline/resources/blue_green.py

```diff
@@ -0,0 +1,224 @@
+        user_parameters = job_data['actionConfiguration']['configuration']['UserParameters']
+        decoded_parameters = json.loads(user_parameters)
+
+    except Exception as e:
+        # We're expecting the user parameters to be encoded as JSON
+        # so we can pass multiple values. If the JSON can't be decoded
+        # then fail the job with a helpful message.
+        raise Exception('UserParameters could not be decoded as JSON')
+
+    if 'blueEnvironment' not in decoded_parameters:
+        # Validate that the stack is provided, otherwise fail the job
+        # with a helpful message.
+        raise Exception('Your UserParameters JSON must include the Blue environment name')
+
+    if 'greenEnvironment' not in decoded_parameters:
+        # Validate that the stack is provided, otherwise fail the job
+        # with a helpful message.
+        raise Exception('Your UserParameters JSON must include the Green environment name')
+
+    if 'application' not in decoded_parameters:
+        # Validate that the artifact name is provided, otherwise fail the job
+        # with a helpful message.
+        raise Exception('Your UserParameters JSON must include the application name')
+
+
+    return decoded_parameters
+
+
+def describe_health(environmentName):
+    response = eb.describe_environment_health(
+    EnvironmentName=environmentName,
+    AttributeNames=[
+        'Status',
+    ]
+    )
+    return response['Status']
+
+
+def describe_app_version(version, application):
+    response = eb.describe_application_versions(
+        ApplicationName=application,
+        VersionLabels=[
+            version,
+        ]
+    )
+    print ("App Version Status: ", response['ApplicationVersions'][0]['Status'])
+    return response['ApplicationVersions'][0]['Status']
+
+def create_app_version(artifact, application):
+    """
+    Creates the application Version to be deployed
+    """
+    bucket = artifact['location']['s3Location']['bucketName']
+    key = artifact['location']['s3Location']['objectKey']
+    version = artifact['revision'] + key
+    version = version.replace("test/MyApp/", "")
+    print ("version: ", version)
+
+    response = eb.create_application_version(
+        ApplicationName=application,
+        VersionLabel=version,
+        SourceBundle={
+            'S3Bucket': bucket,
+            'S3Key': key
+        },
+        Process=True
+    )
+
+    while True:
+        status = describe_app_version(version, application)
+        if status.upper() == "PROCESSED":
+            break
+        else:
+            time.sleep(60)
+    print ("App Version Created.")
+
+    return version
+
+def update_blue_env(blueEnvironment, versionLabel, application):
+    """
+    Deploys to Blue
+    """
+
+    response = eb.update_environment(
+        ApplicationName=application,
+        EnvironmentName=blueEnvironment,
+        VersionLabel=versionLabel,
+    )
+    print ("Blue Deploy result: ",response)
```

Comment:
PEP8 - no space between `print` and `(` and also need a space between `,` and `response`. 

Several of these below.